drivers/t5838/t5838_nrfx_pdm.c

```diff
@@ -28,7 +29,11 @@ BUILD_ASSERT(CONFIG_AUDIO_DMIC_NRFX_PDM == 0,
 
 static void free_buffer(struct t5838_drv_data *drv_data, void *buffer)
 {
+#if KERNELVERSION >= 0x3046300
+	k_mem_slab_free(drv_data->mem_slab, buffer);
+#else
 	k_mem_slab_free(drv_data->mem_slab, &buffer);
+#endif
```

Comment:
We misunderstood each other: 
* use `sys_kernel_version_get` to get the kernel version
* use `SYS_KERNEL_VER_MAJOR` and `SYS_KERNEL_VER_MINOR` macros in a switch statement to determine which api to use.